### PR TITLE
feat: add test case for assetValue and funds

### PR DESCRIPTION
**Pull request by bkioshn**
## Describe your changes
Add test for 
- `utils/funds.ts`
- `utils/assetValue.ts`

Remove 
- `utils/isDecimalNumber.ts`
- `utils/number.ts`

### CODE REVIEW

File: src/lib/utils/assetValue.test.ts [hunk 1]
```diff
@@ -0,0 +1,410 @@
...
+    expect(calTotalValue([])).toEqual(big(0));
+  });
+});
+
+describe("addTokenWithValue", () => {
+  const tokenWithValue1 = {
+    denom: "uadenom",
+    amount: big(100) as U<Token<Big>>,
+    precision: 6,
+    value: big(200) as U<Token<Big>>,
```
Comment (songwongtp) [inline src/lib/utils/assetValue.test.ts:377]: value type is `USD<Big>`

File: src/lib/utils/assetValue.test.ts [hunk 2]
```diff
@@ -0,0 +1,410 @@
...
+    symbol: "",
+    logo: "",
+  } as TokenWithValue;
+
+  const tokenWithValue2 = {
+    denom: "ubdenom",
+    amount: big(100) as U<Token<Big>>,
+    precision: 6,
+    value: big(3500) as U<Token<Big>>,
```
Comment (songwongtp) [inline src/lib/utils/assetValue.test.ts:386]: ditto

File: src/lib/utils/assetValue.test.ts [hunk 3]
```diff
@@ -0,0 +1,410 @@
...
+      tokenWithValue1
+    );
+  });
+
+  test("old value + new value", () => {
+    const expectResult = {
+      denom: "uadenom",
+      amount: big(200) as U<Token<Big>>,
+      precision: 6,
+      value: big(3700) as U<Token<Big>>,
```
Comment (songwongtp) [inline src/lib/utils/assetValue.test.ts:402]: ditto

File: src/lib/utils/assetValue.test.ts [hunk 4]
```diff
@@ -0,0 +1,410 @@
...
+      symbol: "",
+      logo: "",
+    };
+    expect(addTokenWithValue(tokenWithValue1, tokenWithValue2)).toEqual(
```
Comment (songwongtp) [inline src/lib/utils/assetValue.test.ts:406]: also test `addTokenWithValue(tokenWithValue2, tokenWithValue1)`

**Review by songwongtp — CHANGES_REQUESTED** [review]

**Review by evilpeach — CHANGES_REQUESTED** [review]

File: src/lib/utils/assetValue.ts [hunk 5]
```diff
@@ -14,12 +14,15 @@ import type {
...
 import { toToken } from "./formatter";
 
-export const calculateAssetValue = (
+const calculateAssetValue = (
   amount: Token<BigSource>,
   price: USD<number>
 ): USD<Big> => big(amount).mul(price) as USD<Big>;
 
 export const calAssetValueWithPrecision = (balance: Balance): USD<Big> => {
+  if (Number.isNaN(Number(balance.amount)) || !balance.amount.length)
```
Comment (evilpeach) [inline src/lib/utils/assetValue.ts:23]: Please add the `trim()` function and also introduce a test case for when `balance.amount` is a space.

**Review by evilpeach — APPROVED** [review]

**Pull request merged**